src/core/mempool.rs

```diff
@@ -1175,71 +1249,137 @@ impl MemPoolDB {
+                    let opt_tx = if start_with_no_estimate {
+                        null_iterator
+                            .next()
+                            .map_err(|err| Error::SqliteError(err))?
+                    } else {
+                        fee_iterator.next().map_err(|err| Error::SqliteError(err))?
+                    };
+                    match opt_tx {
+                        Some(row) => (MemPoolTxInfoPartial::from_row(row)?, start_with_no_estimate),
+                        None => {
+                            // If the selected iterator is empty, check the other
+                            match if start_with_no_estimate {
+                                fee_iterator.next().map_err(|err| Error::SqliteError(err))?
+                            } else {
+                                null_iterator
+                                    .next()
+                                    .map_err(|err| Error::SqliteError(err))?
+                            } {
+                                Some(row) => (
+                                    MemPoolTxInfoPartial::from_row(row)?,
+                                    !start_with_no_estimate,
+                                ),
+                                None => {
+                                    debug!("No more transactions to consider in mempool");
+                                    break;
+                                }
+                            }
                         }
-                        let min_nonce =
-                            StacksChainState::get_account(clarity_tx, &address.clone().into())
-                                .nonce;
-
-                        self.update_last_known_nonces(&address, min_nonce)?;
-                        last_addr = Some(address)
                     }
                 }
-                ConsiderTransactionResult::Consider(consider) => {
-                    // if we actually consider the chosen transaction,
-                    //  compute a new start_with_no_estimate on the next loop
-                    remember_start_with_estimate = None;
-                    debug!("Consider mempool transaction";
+            };
+
+            // Check the nonces.
+            let expected_origin_nonce = nonce_cache.get(&candidate.origin_address, clarity_tx);
+            let expected_sponsor_nonce = nonce_cache.get(&candidate.sponsor_address, clarity_tx);
+            match order_nonces(
+                candidate.origin_nonce,
+                expected_origin_nonce,
+                candidate.sponsor_nonce,
+                expected_sponsor_nonce,
+            ) {
+                Ordering::Less => {
+                    debug!(
+                        "Mempool: unexecutable: drop tx ({})",
+                        candidate.fee_rate.unwrap_or_default()
+                    );
+                    // This transaction cannot execute in this pass, just drop it
```

Comment:
Since we're doing this check in multiple passes, and since a transaction can go from having the `::Greater` condition, to having the `::Equal` condition, to having the `::Less` condition, would it make sense here to also evict state from the nonce cache for this candidate?